v2/pkg/catalog/loader/loader.go

```diff
@@ -267,7 +267,11 @@ func (store *Store) LoadTemplates(templatesList []string) []*templates.Template
 				stats.Increment(parsers.RuntimeWarningsStats)
 				gologger.Warning().Msgf("Could not parse template %s: %s\n", templatePath, err)
 			} else if parsed != nil {
-				loadedTemplates = append(loadedTemplates, parsed)
+				if len(parsed.RequestsHeadless) > 0 && !store.config.ExecutorOptions.Options.Headless {
+					gologger.Warning().Msgf("Headless flag is required for headless templates %s\n", templatePath)
```

Comment:
As we are handling one template at a time, what about something like:
```go
gologger.Warning().Msgf("Headless flag is required for headless template: %s\n", templatePath)
```

Reply:
I agree.